### PR TITLE
Adds node hooks: pre, mid and post update hook.

### PR DESCRIPTION
**Pull request by jtudelag**
mid hook after updating the node, but before making it schedulable again.

### CODE REVIEW

**Comment by sdodson** [comment]
bot, test pull request

File: playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml [hunk 1]
```diff
@@ -46,10 +46,36 @@
     - l_upgrade_nodes_drain_result is failed
     - openshift_upgrade_nodes_drain_timeout | default(0) | int == 0
 
+  # Run the pre-upgrade hook if defined:
+  - debug: msg="Running node pre-upgrade hook {{ openshift_node_upgrade_pre_hook }}"
+    when: openshift_node_upgrade_pre_hook is defined
+
```
Comment (DanyC97) [inline playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml]: would be nicer if we use the asset module?

File: playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml [hunk 2]
```diff
@@ -46,10 +46,36 @@
...
   post_tasks:
   - import_role:
       name: openshift_node
       tasks_from: upgrade.yml
+
+  # Run the upgrade hook prior to make the node schedulable again.
+  - debug: msg="Running node upgrade hook {{ openshift_node_upgrade_hook }}"
+    when: openshift_node_upgrade_hook is defined
+
```
Comment (DanyC97) [inline playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml]: likewise here, assert module?

File: playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml [hunk 3]
```diff
@@ -63,3 +89,10 @@
       name: openshift_excluder
     vars:
       r_openshift_excluder_action: enable
+
+  # Run the post-upgrade hook if defined:
+  - debug: msg="Running node post-upgrade hook {{ openshift_node_upgrade_post_hook }}"
+    when: openshift_node_upgrade_post_hook is defined
```
Comment (DanyC97) [inline playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml]: same here, assert module

File: playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml (hunk 1 above)
Comment (sdodson) [reply playbooks/common/openshift-cluster/upgrades/upgrade_nodes.yml]: How would you suggest the assert module is used here? Do you mean asserting that the referenced task file exists?

**Pull request merged**